crates/core/src/client/mod.rs

```diff
@@ -701,7 +664,7 @@ impl<P: Provider + Send + Sync + 'static> KakarotStarknetApi<P> for KakarotClien
         block_number: Option<U256>,
     ) -> BlockTransactions {
         let handles = Into::<Vec<TransactionType>>::into(initial_transactions).into_iter().map(|tx| async move {
-            let tx = Into::<StarknetTransaction>::into(tx);
+            let tx = Into::<StarknetBlockTransaction>::into(tx);
```

Comment:
Why this name?

Reply:
so I need to distinguish a block transaction i.e. a tx that has been accepted by the block, and a signed transaction, i.e. a transaction that is signed and sent to a sequencer for inclusion in the block.

Reply:
then i believe this is a misleading name, i suggest:
`IncludedStarknetTransaction` or `AcceptedStarknetTransaction` vs. `PendingStarknetTransaction`, `SignedStarknetTransaction` or perhaps simply `StarknetTransaction`

Reply:
You think StarknetBlockTransaction is misleading? Otherwise I just keep the starknet rs naming, which is transaction for block transaction and broacasted transaction for signed

Reply:
Or transaction vs signed transaction as reth

Reply:
> Or transaction vs signed transaction as reth

this is fine imo